awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,259 @@
-import React, { useEffect, useCallback } from 'react';
+import React, { useEffect, useCallback, useState } from 'react';
 import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 
-import useRequest from '@util/useRequest';
+import useRequest, { useDeleteItems } from '@util/useRequest';
+import { Switch, Checkbox, Button } from '@patternfly/react-core';
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarGroup,
+  DataToolbarItem,
+} from '@patternfly/react-core/dist/umd/experimental';
 import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
+import ErrorDetail from '@components/ErrorDetail';
 import { JobTemplatesAPI } from '@api';
+import ContentEmpty from '@components/ContentEmpty';
+import { ToolbarAddButton } from '@components/PaginatedDataList';
+import { getQSConfig } from '@util/qs';
 import SurveyListItem from './SurveyListItem';
 
-function SurveyList({ template }) {
+import AlertModal from '../../../components/AlertModal';
+
+const QS_CONFIG = getQSConfig('survey', {
+  page: 1,
+});
+
+function SurveyList({ template, i18n }) {
+  const [selected, setSelected] = useState([]);
+  const [surveyEnabled, setSurveyEnabled] = useState(template.survey_enabled);
+  const [showToggleError, setShowToggleError] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
   const {
-    result: questions,
+    result: { questions, name, description },
     error: contentError,
     isLoading,
     request: fetchSurvey,
   } = useRequest(
     useCallback(async () => {
       const {
-        data: { spec },
+        data: { spec, description: surveyDescription, name: surveyName },
       } = await JobTemplatesAPI.readSurvey(template.id);
-
-      return spec.map((s, index) => ({ ...s, id: index }));
-    }, [template.id])
+      return {
+        questions: spec?.map((s, index) => ({ ...s, id: index })),
+        description: surveyDescription,
+        name: surveyName,
+      };
+    }, [template.id]),
+    { questions: [], name: '', description: '' }
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
-  if (contentError) {
-    return <ContentError error={contentError} />;
-  }
-  if (isLoading) {
-    return <ContentLoading />;
-  }
 
-  return (
-    questions?.length > 0 &&
-    questions.map((question, index) => (
+  const isAllSelected =
+    selected.length === questions?.length && selected.length > 0;
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: deleteQuestions,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      if (isAllSelected) {
+        return JobTemplatesAPI.destroySurvey(template.id);
+      }
+      const surveyQuestions = [];
+      questions.forEach(q =>
+        !selected.find(s => s.id === q.id) ? surveyQuestions.push(q) : null
```

Comment:
`foo = bar ? : fiz` is a conditional assignment. You generally should only use it for assignment statements.

`surveyQuestions.push(q)` is not an assignment. It would be more clear to do this with a regular old if statement:

```jsx
questions.forEach(q =>
  if (!selected.some(s => s.id === q.id)) {
    surveyQuestions.push(q)
  }
});
```

The other change you might notice is that we're using `some` instead of `find`.

`find` returns the first value that matches your filter. You're not actually using the returned value for anything, though.

, `some` returns a true/false based on whether or not some value exists that matches your filter, which I think is what you want.

Reply:
doesn't `some` stop as soon as it returns true, though?  What if the user is deleting more than 1 question.

Reply:
:man_shrugging: maybe? Array.find also "stops" at (returns) the first item it finds that matches the filter.

What do you want this function to do? Which questions should `surveyQuestions` actually contain?

Reply:
When I say it's changed I mean I changed the code to use `some` instead of `find`.  This function should return the questions that are not selected so that they can then be sent to the api, thus removing those that were selected. It's sort of opposite of all of our other delete patterns.